cli/src/main.rs

```diff
@@ -659,53 +659,125 @@ fn run<I: IntoIterator<Item = T>, T: Into<OsString> + Clone>(args: I) -> Result<
-                    .arg(
-                        Arg::with_name("private_key_file")
-                            .value_name("private-key-file")
-                            .short("k")
-                            .long("key")
-                            .takes_value(true)
-                            .help("Name or path of private key"),
-                    ),
+                ))
+                .arg(
+                    Arg::with_name("force")
+                        .long("force")
+                        .help("Overwrite node if it already exists"),
+                )
+                .arg(
+                    Arg::with_name("status_url")
+                        .takes_value(true)
+                        .help("URL of splinter REST API to query for node data"),
+                )
+                .arg(
+                    Arg::with_name("key_files")
+                        .long("key-file")
+                        .takes_value(true)
+                        .multiple(true)
+                        .required(true)
+                        .help("Path of public key file to include with node"),
+                )
+                .arg(
+                    Arg::with_name("metadata")
+                        .long("metadata")
+                        .takes_value(true)
+                        .multiple(true)
+                        .help("Metadata to include with node (<key>=<value>)"),
+                )
+                .arg(
+                    Arg::with_name("private_key_file")
+                        .value_name("private-key-file")
+                        .short("k")
+                        .long("key")
+                        .takes_value(true)
+                        .help("Name or path of private key"),
+                ),
+        );
+
+    #[cfg(feature = "registry")]
+    let registry_command = registry_command.subcommand(
+        SubCommand::with_name("add")
+            .about("Add a node to the local registry")
+            .arg(
+                Arg::with_name("display_name")
+                    .long("display-name")
+                    .takes_value(true)
+                    .help("Human-readable name for the new node"),
+            )
+            .arg(
+                Arg::with_name("dry_run")
+                    .long("dry-run")
+                    .help("Show the expected changes without submitting the node"),
+            )
+            .arg(
+                Arg::with_name("endpoint")
+                    .long("endpoint")
+                    .takes_value(true)
+                    .multiple(true)
+                    .required_unless("from_remote")
+                    .help("Network endpoint for the new node"),
+            )
+            .arg(
+                Arg::with_name("from_remote")
+                    .long("from-remote")
+                    .conflicts_with_all(&[
+                        "display_name",
+                        "endpoint",
+                        "identity",
```

Comment:
`identity` is still required with `--from-remote` since it determines which node to copy. I just realized that I said to include this in my comment above, so that's 100% my mistake.

Reply:
Fixed